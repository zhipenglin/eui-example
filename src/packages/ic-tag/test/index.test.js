import React,{Fragment} from 'react';
import IcTag from '../src/index.js';
import renderer from 'react-test-renderer';

jest.mock('ic-iconfont',()=>'mock-icon');

it('more than seven', () => {
    const tree = renderer.create(<Fragment>
        <IcTag className="ic-tag-wrapper" closable={true} >我是大于七个字的可关闭标签</IcTag>
        <IcTag className="ic-tag-wrapper" color="#ffa000" >我是大于七个字的关键词标签</IcTag>
    </Fragment>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('less than four', () => {
    const tree = renderer.create(<Fragment>
        <IcTag className="ic-tag-wrapper" closable={true} >四以下</IcTag>
        <IcTag className="ic-tag-wrapper" color="#ffa000" >四以下</IcTag>
    </Fragment>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('four', () => {
    const tree = renderer.create(<Fragment>
        <IcTag className="ic-tag-wrapper" closable={true} >四字标签</IcTag>
        <IcTag className="ic-tag-wrapper" color="#ffa000" >四字标签</IcTag>
    </Fragment>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('five', () => {
    const tree = renderer.create(<Fragment>
        <IcTag className="ic-tag-wrapper" closable={true} >五个字标签</IcTag>
        <IcTag className="ic-tag-wrapper" color="#ffa000" >五个字标签</IcTag>
    </Fragment>).toJSON();
    expect(tree).toMatchSnapshot();
});
