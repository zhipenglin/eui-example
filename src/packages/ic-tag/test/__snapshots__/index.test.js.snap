// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`five 1`] = `
Array [
  <div
    className="ic-tag ic-tag-wrapper ic-close-tag"
    onClose={[Function]}
    style={
      Object {
        "background": "RGB(255,160,0,0.08)",
        "color": "#FFA000",
      }
    }
  >
    <span
      className="ic-tag-inner"
    >
      五个字标签
    </span>
    <mock-icon
      className="ic-tag-icon"
      onClick={[Function]}
      size={15}
      type="close_small"
    />
  </div>,
  <div
    className="ic-tag ic-tag-wrapper"
    onClose={[Function]}
    style={
      Object {
        "background": "RGB(255,160,0,0.08)",
        "color": "#ffa000",
      }
    }
  >
    <span
      className="ic-tag-inner"
    >
      五个字标签
    </span>
  </div>,
]
`;

exports[`four 1`] = `
Array [
  <div
    className="ic-tag ic-tag-wrapper ic-close-tag"
    onClose={[Function]}
    style={
      Object {
        "background": "RGB(255,160,0,0.08)",
        "color": "#FFA000",
      }
    }
  >
    <span
      className="ic-tag-inner"
    >
      四字标签
    </span>
    <mock-icon
      className="ic-tag-icon"
      onClick={[Function]}
      size={15}
      type="close_small"
    />
  </div>,
  <div
    className="ic-tag ic-tag-wrapper"
    onClose={[Function]}
    style={
      Object {
        "background": "RGB(255,160,0,0.08)",
        "color": "#ffa000",
      }
    }
  >
    <span
      className="ic-tag-inner"
    >
      四字标签
    </span>
  </div>,
]
`;

exports[`less than four 1`] = `
Array [
  <div
    className="ic-tag ic-tag-wrapper ic-close-tag"
    onClose={[Function]}
    style={
      Object {
        "background": "RGB(255,160,0,0.08)",
        "color": "#FFA000",
      }
    }
  >
    <span
      className="ic-tag-inner"
    >
      四以下
    </span>
    <mock-icon
      className="ic-tag-icon"
      onClick={[Function]}
      size={15}
      type="close_small"
    />
  </div>,
  <div
    className="ic-tag ic-tag-wrapper"
    onClose={[Function]}
    style={
      Object {
        "background": "RGB(255,160,0,0.08)",
        "color": "#ffa000",
      }
    }
  >
    <span
      className="ic-tag-inner"
    >
      四以下
    </span>
  </div>,
]
`;

exports[`more than seven 1`] = `
Array [
  <div
    className="ic-tag ic-tag-wrapper ic-close-tag"
    onClose={[Function]}
    style={
      Object {
        "background": "RGB(255,160,0,0.08)",
        "color": "#FFA000",
      }
    }
  >
    <span
      className="ic-tag-inner"
    >
      我是大于七个字的可关闭标签
    </span>
    <mock-icon
      className="ic-tag-icon"
      onClick={[Function]}
      size={15}
      type="close_small"
    />
  </div>,
  <div
    className="ic-tag ic-tag-wrapper"
    onClose={[Function]}
    style={
      Object {
        "background": "RGB(255,160,0,0.08)",
        "color": "#ffa000",
      }
    }
  >
    <span
      className="ic-tag-inner"
    >
      我是大于七个字的关键词标签
    </span>
  </div>,
]
`;
